cli/src/index.ts

```diff
@@ -0,0 +1,61 @@
+import { program, Option } from 'commander';
+import Upload from './commands/upload';
+import ServerInfo from './commands/server-info';
+import LoginKey from './commands/login/key';
+
+program.name('immich').description('Immich command line interface');
+
+program
+  .command('upload')
+  .description('Upload assets')
+  .usage('[options] [paths...]')
+  .addOption(new Option('-r, --recursive', 'Recursive').env('IMMICH_RECURSIVE').default(false))
+  .addOption(new Option('-i, --ignore [paths...]', 'Paths to ignore').env('IMMICH_IGNORE_PATHS'))
+  .addOption(new Option('-h, --skip-hash', "Don't hash files before upload").env('IMMICH_SKIP_HASH').default(false))
+  .addOption(
+    new Option('-n, --dry-run', "Don't perform any actions, just show what will be done")
+      .env('IMMICH_DRY_RUN')
+      .default(false),
+  )
+  .addOption(new Option('--delete', 'Delete local assets after upload').env('IMMICH_DELETE_ASSETS'))
+  .argument('[paths...]', 'One or more paths to assets to be uploaded')
+  .action((paths, options) => {
+    options.excludePatterns = options.ignore;
+    new Upload().run(paths, options);
+  });
+
+program
+  .command('import')
+  .description('Import existing assets')
+  .usage('[options] [paths...]')
+  .addOption(new Option('-r, --recursive', 'Recursive').env('IMMICH_RECURSIVE').default(false))
```

Comment:
There are a decent amount of shared options between these two. Any way to define them once and use them in both places? Or have a function that can be called from both places to add them in one place?

Reply:
There's no easy way to do it, I decided that the cleanest way is the current one